### PR TITLE
Sa li/dev/fr bc/add confirm and show password on register page

### PR DESCRIPTION
**Pull request by theSaLi**
Add confirm and show password on register page.
Had to use <InputGroup> to add the show password button.
Add a logic to validate the username and password.
Please review, thanks.
image

### CODE REVIEW

File: frontend/src/components/registerview/RegisterView.jsx [hunk 1]
```diff
@@ -1,50 +1,87 @@
...
 class RegisterView extends Component {
-  constructor(props){
+  constructor(props) {
     super(props);
     this.state = {
       username: '',
-      password: ''
+      password: '',
+      confirm_password: '',
+      password_type: 'password'
```
Comment (changdiqing) [inline frontend/src/components/registerview/RegisterView.jsx:14]: - [ ] Comparing password_type as string looks like an antipattern to me. If we have only two statuses, I would suggest using boolean i.e. state.passwordVisible, and check if it is true. If we have more than 2 statuses, I would create an **enum** to store them. One way to use enum in JS is to simply [create a JS object and freeze it](https://stackoverflow.com/questions/287903/what-is-the-preferred-syntax-for-defining-enums-in-javascript)

**Review by changdiqing — CHANGES_REQUESTED** [review]
Looks pretty good to me, I left one suggestion as comment.

**Comment by theSaLi** [comment]
Modified as suggested, please review again. Thanks.

**Comment by Apollo1840** [comment]
@theSaLi There is no modification that can be observed from my side, maybe push it again?

**Pull request merged**